iterators.tex

```diff
@@ -7346,23 +7352,287 @@
 
 \indexlibrary{\idxcode{crbegin(const C\& c)}}%
 \begin{itemdecl}
-template <@\oldtxt{class}\newtxt{_Auto}@ C> auto crbegin(const C& c) -> decltype(@\oldoldtxt{std::}\newtxt{ranges\oldtxt{_v1}::}@rbegin(c));
+template <class C> auto crbegin(const C& c) -> decltype(std::rbegin(c));
 \end{itemdecl}
 \begin{itemdescr}
-\pnum \returns \tcode{\oldoldtxt{std::}\newtxt{ranges\oldtxt{_v1}::}rbegin(c)}.
+\pnum \returns \tcode{std::rbegin(c)}.
 \end{itemdescr}
 
 \indexlibrary{\idxcode{crend(const C\& c)}}%
 \begin{itemdecl}
-template <@\oldtxt{class}\newtxt{_Auto}@ C> auto crend(const C& c) -> decltype(@\oldoldtxt{std::}\newtxt{ranges\oldtxt{_v1}::}@rend(c));
+template <class C> auto crend(const C& c) -> decltype(std::rend(c));
 \end{itemdecl}
 \begin{itemdescr}
-\pnum \returns \tcode{\oldoldtxt{std::}\newtxt{ranges\oldtxt{_v1}::}rend(c)}.
+\pnum \returns \tcode{std::rend(c)}.
 \end{itemdescr}
+}
+
+{\color{newclr}
+\rSec2[iterator.range.begin]{\tcode{begin}}
+\pnum
+The name \tcode{begin} denotes a customization point object
+(\ref{customization.point.object}). The expression
+\tcode{ranges::begin(E)} for a subexpression \tcode{E} has the same
+effects as:
+
+\begin{itemize}
+\item
+  \tcode{(E).begin()} if \tcode{E} has type
+  \tcode{std::initializer_list<T>} for some type \tcode{T}.
+
+\item
+  Otherwise, \tcode{ranges::begin((const T\&)(E))} if \tcode{E} is an
+  rvalue of some type \tcode{T}. This usage is deprecated.
+  \enternote This deprecated usage exists so that
+  \tcode{ranges::begin(E)} behaves similarly to \tcode{std::begin(E)}
+  as defined in C++14 when \tcode{E} is an rvalue. \exitnote
+
+\item
+  Otherwise, \tcode{(E) + 0} if \tcode{E} has array type.
+
+\item
+  Otherwise, \tcode{(E).begin()} if it is a valid expression whose
+  type \tcode{I} meets the syntactic requirements of \tcode{Iterator<I>}.
+  If \tcode{Iterator} is not satisfied, the program is ill-formed
+  with no diagnostic required.
+
+\item
+  Otherwise, \tcode{begin(E)} if it is a valid expression whose type
```

Comment:
"if it is a valid expression" -> would probably be covered by the blanket effects-equivalent-to above.